src/pages/workspaces/workspace/WorkspaceContainer.js

```diff
@@ -79,7 +80,8 @@ const AzureWarning = () => {
 }
 
 
-const GooglePermissionsWarning = () => {
+const GooglePermissionsWarning = ({ workspace }) => {
+  Ajax().Metrics.captureEvent(Events.permissionsSynchronizationDelayDisplayed, extractWorkspaceDetails(workspace))
```

Comment:
I think this may cause a LOT of events to fire, since I believe this will be called on every render. Did you check the number of events being sent?